lib/config/param.c

```diff
@@ -921,6 +973,15 @@ param_default_validator(const struct param_spec *ps, const void *value)
                 }                                                                                 \
                 *(int64_t *)value = (uint64_t)tmp;                                                \
                 break;                                                                            \
+            case PARAM_TYPE_SIZE:                                                                 \
+                if (tmp < 0 || tmp > (double)SIZE_MAX) {                                          \
+                    log_err(                                                                      \
+                        "Number of bytes of %s is not within the bounds of a size_t",             \
+                        ps->ps_name);                                                             \
+                    return false;                                                                 \
+                }                                                                                 \
+                *(size_t *)value = (size_t)tmp;                                                   \
+                break;                                                                            \
```

Comment:
I'm suprised the compiler doesn't complain about lost of precision due to cast to double...

Reply:
Perhaps it would if we had `-Wconversion` enabled

Reply:
Casting to (long double) should preserve all 64 bits, I think...